index.Dockerfile

```diff
@@ -0,0 +1,15 @@
+FROM quay.io/operator-framework/upstream-opm-builder:v1.19.5 AS builder
```

Comment:
Can you pls tell me why we need this, What is the motive behind adding the custom dockerfile instead of using the OPM binary directly? Also as we are not using OPM anymore we don't need opm target anymore. 

But I would still prefer using the opm binary instead of creating a dockerfile

Reply:
Specifically, this Dockerfile *already contains* the OPM binary. By using a container image, we completely avoid the problem of ensuring we have the correct version of the binary on the developer's system and all the directory path logic to keep it in the right place.

But you raise an interesting point. If we're not using OPM then what are we using to generate the CatalogSource index image? It's cool if it exists, I'm just not aware of any alternative.

Reply:
Jose, I guess you misunderstood my point I mean to say is we should also remove a target in the make file which downloads the opm binary [1]. Also, we should stop calling the target in the catalog-build target [2].

[1]. https://github.com/red-hat-storage/odf-operator/blob/main/hack/make-tools.mk#L27-L30
[2]. https://github.com/red-hat-storage/odf-operator/blob/main/Makefile#L165